### PR TITLE
WT-11151 Merge code coverage competition entries into coverage-report task

### PR DESCRIPTION
**Pull request by jeremythorp**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -3430,7 +3429,183 @@ tasks:
             # Record the start time, in seconds
             date +%s > ../time.txt
             # Perform the tests to get code coverage metrics for
```
Comment (lukech) [inline test/evergreen.yml:3431]: It'd be nice to have a comment describing the philosophy/criteria used for the test selection (for our future selves).

File: test/evergreen.yml [hunk 2]
```diff
@@ -3430,7 +3429,183 @@ tasks:
...
+            . ../test/evergreen/find_cmake.sh
+            
             ${test_env_vars|} test/unittest/unittests
+
+            ${test_env_vars|} $CTEST -R '(ex_|lsm_workload|filesys|metadata_salvage|col_efficiency|dup_index_collator|compact_quick|test_checkpoint_4_mixed_timestamps|test_checkpoint_6_row_named_prepare|random_directio|scope|join_main_table_row|pad_byte_collator|meta_ckptlist_get_alloc)'
+
+            ${test_env_vars|} /opt/mongodbtoolchain/v4/bin/python3 ../test/suite/run.py test_rollback_to_stable03
```
Comment (lukech) [inline test/evergreen.yml:3438]: Could `/opt/mongodbtoolchain/v4/bin/python3` be replaced by `${python_binary|python3}`, the same as some lines in below?

File: test/evergreen.yml [hunk 3]
```diff
@@ -3430,7 +3429,183 @@ tasks:
...
+            ${test_env_vars|} ./test/csuite/wt4156_metadata_salvage/test_wt4156_metadata_salvage
+            ${test_env_vars|} ./test/csuite/wt2403_lsm_workload/test_wt2403_lsm_workload
+
+            ${test_env_vars|} WIREDTIGER_HOME=ex1 examples/c/ex_all/ex_all & ${test_env_vars|} WIREDTIGER_HOME=ex2 examples/c/ex_schema/ex_schema
+            # Rest of the examples
+            cd examples/c/
+            ${test_env_vars|} $CTEST --output-on-failure 2>&1
+            cd ../../
+
+            # Faster csuite tests. 30% in ~1 minute
```
Comment (lukech) [inline test/evergreen.yml:3467]: Does "30%" indicate the code coverage % the below set of tests is able to achieve?

File: test/evergreen.yml [hunk 4]
```diff
@@ -3430,7 +3429,183 @@ tasks:
...
+            ${test_env_vars|} $(pwd)/test/csuite/wt4699_json/test_wt4699_json 2>&1
+            ${test_env_vars|} $(pwd)/test/csuite/wt4891_meta_ckptlist_get_alloc/test_wt4891_meta_ckptlist_get_alloc 2>&1
+            ${test_env_vars|} $(pwd)/test/csuite/wt9937_parse_opts/test_wt9937_parse_opts 2>&1
+            ${test_env_vars|} $(pwd)/test/csuite/wt3363_checkpoint_op_races/test_wt3363_checkpoint_op_races 2>&1
+            ${test_env_vars|} $(pwd)/test/csuite/wt3120_filesys/test_wt3120_filesys -b $(pwd) 2>&1
+            ${test_env_vars|} $(pwd)/test/csuite/wt8659_reconstruct_database_from_logs/test_wt8659_reconstruct_database_from_logs 2>&1
+
+            # A subset of python tests that cover areas not touched by csuite
+            # In some cases use -s 1 to only run the first scenario for each test
+            # (To check. Some of these can probably be removed)
```
Comment (lukech) [inline test/evergreen.yml:3487]: Curious if it's an obsolete comment.

File: test/suite/configfile.cc
Comment (lukech) [inline test/suite/configfile.cc:1]: It does not look like this config file is actively used. I believe the config file can be used to skip certain long-running Python tests to bring down the overall test running time.

File: test/evergreen.yml (hunk 3 above)
Comment (jeremythorp) [reply test/evergreen.yml:3467]: That comment was copy/pasted from the original competition entry, and is now obsolete.

File: test/evergreen.yml (hunk 4 above)
Comment (jeremythorp) [reply test/evergreen.yml:3487]: That's correct - that comment was copy/pasted from the original competition entry, and is now obsolete. I'll remove it.

File: test/evergreen.yml (hunk 2 above)
Comment (jeremythorp) [reply test/evergreen.yml:3438]: Good idea. That was copy/pasted from the original competition entry and should be updated.

Comment (jeremythorp) [reply test/evergreen.yml]: I've updated that now.

File: test/suite/configfile.cc
Comment (jeremythorp) [reply test/suite/configfile.cc:1]: I used the config file initially, but then removed the test that was using it to reduce overlapping tests. I've removed it for now, and we can return it later if needed.

**Review by lukech — APPROVED** [review]
LGTM

**Pull request merged**